server/channels/api4/group.go

```diff
@@ -631,15 +625,47 @@ func unlinkGroupSyncable(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func verifyLinkUnlinkPermission(c *Context, syncableType model.GroupSyncableType, syncableID string) *model.AppError {
+	group, appErr := c.App.GetGroup(c.Params.GroupId, nil, nil)
+	if appErr != nil {
+		return appErr
+	}
+
+	if group.Source != model.GroupSourceLdap {
+		return model.NewAppError("Api4.linkGroupSyncable", "app.group.crud_permission", nil, "", http.StatusBadRequest)
+	}
+
+	// If AllowReference is disabled, limit who can link the group.
+	// This voids leaking the list of group members.
+	// See https://mattermost.atlassian.net/browse/MM-55314 for more details.
+	if !group.AllowReference {
```

Comment:
Since this operation is available only via system console, why are we checking AllowReference here instead of checking the permission regardless of the allow reference? From a UI perspective, someome without Groups system console access has no way to link/unlink groups to channels/teams.

Reply:
It is correct that the UI only allows users with access to the groups' system console to link groups.

I still think this check is correct, as it aligns with the changes in https://github.com/mattermost/mattermost/pull/26551. It keeps the permission checks consistent with each other, even though the UI doesn't support linking for non-admins.

Reply:
Linking/unlinking does more than just fetching the user groups though. I don't think that the changes in https://github.com/mattermost/mattermost/pull/26551 are related here. In that PR it was about fetching, here it's about linking.

Reply:
The original issue (https://mattermost.atlassian.net/browse/MM-55314) only affects groups where AllowReference is disabled. Requiring `PermissionSysconsoleWriteUserManagementGroups` to link a group, even if AllowReference is enabled, feels out of the scope of the originally reported issue.

I see that linking a group would count as modifying it. But this check protects the member list, not the state of the group.

Reply:
Let's ignore my initial comment. If someone can invite users to a team and the group can be referenced, it makes sense that they can add it.